client/src/index.jsx

```diff
@@ -53,29 +49,65 @@ class App extends React.Component {
   }
 
   calculatePayments() {
-    const homePrice = this.state.currentValues.home_price;
-    const propertyTax = this.state.currentValues.property_tax;
-    const homeInsurance = this.state.currentValues.home_insurance;
-    const hoaDues = this.state.currentValues.hoa_dues;
-    const downPaymentAmount = this.state.currentValues.down_payment_amount;
-    const interestRate = this.state.currentValues.interest_rate;
-
-    const loanAmount = homePrice - downPaymentAmount;
-    const firstMonthPrincipal = Math.floor((loanAmount * 1.2 / 100) / 12);
-    const firstMonthInterest = Math.floor((loanAmount * interestRate / 100) / 12);
-    const firstYearPandI = firstMonthPrincipal + firstMonthInterest;
-    const monthlyPayment = firstYearPandI + propertyTax + homeInsurance + hoaDues;
+    let homePrice = this.state.currentValues.home_price;
+    let propertyTax = Math.round(this.state.currentValues.property_tax/12);
+    let homeInsurance = Math.round(this.state.currentValues.home_insurance/12);
+    let hoaDues = this.state.currentValues.hoa_dues;
+    let downPaymentAmount = this.state.currentValues.down_payment_amount;
+    let interestRate = this.state.currentValues.interest_rate;
+    let years = this.state.currentValues.years;
+    let downPaymentPercentage = this.state.currentValues.down_payment_percentage;
+
+    let loanAmount = homePrice - downPaymentAmount;
+    let perMonthInterest = interestRate/1200;
+    let numberOfPayments = years * 12;
+    let monthlyPandI = 0;
+
+    if(interestRate > 0) {
+      monthlyPandI = loanAmount * (perMonthInterest * (Math.pow((1 + perMonthInterest), numberOfPayments))) / (Math.pow((1 + perMonthInterest), numberOfPayments) - 1);
+    }else{
+      monthlyPandI = loanAmount/numberOfPayments;
+    }
+    let roundedMonthlyPandI = Math.round(monthlyPandI);
+    let newPMI = 0;
+
+    if(downPaymentPercentage < 20 && this.state.checked.pmiChecked){
```

Comment:
space between parens
```suggestion
    if (downPaymentPercentage < 20 && this.state.checked.pmiChecked) {
```